src/throughput.rs

```diff
@@ -0,0 +1,115 @@
+use std::fmt::Debug;
+use std::fmt::{self, Display};
+use std::time::Duration;
+use std::time::Instant;
+
+/// Entry for throughput monitor with measured information.
+struct ThroughputEntry {
+    measured_throughput: u32,
+    start: Instant,
+}
+
+impl ThroughputEntry {
+    /// Construct a new throughput entry.
+    pub fn new(measured_throughput: u32, time: Instant) -> ThroughputEntry {
+        ThroughputEntry {
+            measured_throughput,
+            start: time,
+        }
+    }
+}
+
+/// Helper to monitor throughput.
+///
+/// Throughput is calculated at some duration.
+/// For each duration an entry is created to keep track of the history of throughput.
+///
+/// With this type you can calculate the average or get the total and last measured throughput.
+pub struct ThroughputMonitoring {
+    throughput_duration: Duration,
+    timer: Instant,
+    current_throughput: u32,
+    measured_throughput: Vec<ThroughputEntry>,
```

Comment:
I found this name to be very confusing. Can we make it something like "throughput_measurements"?

Reply:
To me, it seems the same, I guess it better, let's go for it